### PR TITLE
LibGfx+Utilities: Add animation utility, make it write animated webps

### PR DESCRIPTION
**Pull request by nico**
The high-level design is that we have a static method on WebPWriter that
returns an AnimationWriter object. AnimationWriter has a virtual method
for writing individual frames. This allows streaming animations to disk,
without having to buffer up the entire animation in memory first.
The semantics of this function, add_frame(), are that data is flushed
to disk every time the function is called, so that no explicit `close()`
method is needed.

For some formats that store animation length at the start of the file,
including WebP, this means that this needs to write to a SeekableStream,
so that add_frame() can seek to the start and update the size when a
frame is written.

This design should work for GIF and APNG writing as well. We can move
AnimationWriter to a new header if we add writers for these.

Currently, `animation` can read any animated image format we can read
(apng, gif, webp) and convert it to an animated webp file.

The written animated webp file is not compressed whatsoever, so this
creates large output files at the moment.

### CODE REVIEW

**Comment by nico** [comment]
I'm not married to adding an `animation` tool – maybe `image` should be able to do this, and there a thousand other alternative viable approaches. For now, this is simple and works, and I'm more interested in the writing code than in the tool I invoke to ultimately call that code :)

Here's

```sh
% curl -O 'https://media2.giphy.com/media/v1.Y2lkPTc5MGI3NjExOHhwZzRsMGV0bm83dmtjY2VnMmdiZmtybnhoM3phdGl0ejJja3RtbyZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/XT9HMdwmpHqqOu1f1a/giphy.gif'    
% Build/lagom/bin/animation -o wow.webp giphy.gif
```

(the .gif is 184K, the .webp is 21M (!))

…hm, can't attach webp files on github yet, apparently! https://github.com/orgs/community/discussions/5470

Well, it looks like the gif, except that it's 114 times is size. (Again, ahis is due to our webp encoder not yet doing any compression. It's a 480 x 480 image with 24 frames. At 4 bytes per pixel, that's 22MB / 21 MiB.)

**Comment by nico** [comment]
Red "Lagom / CI (macOS, macos-14) (pull_request)" is unrelated.

**Review by LucasChollet — APPROVED** [review]
LGTM!

File: Userland/Libraries/LibGfx/ImageFormats/WebPWriter.cpp [hunk 1]
```diff
@@ -327,4 +327,192 @@ ErrorOr<void> WebPWriter::encode(Stream& stream, Bitmap const& bitmap, Options c
...
+        TRY(stream.write_value<u8>(0));
+    return {};
+}
+
+struct ANMFChunk {
+    u32 frame_x;
+    u32 frame_y;
+    u32 frame_width;
+    u32 frame_height;
+    u32 frame_duration_in_milliseconds;
```
Comment (trflynn89) [inline Userland/Libraries/LibGfx/ImageFormats/WebPWriter.cpp:370]: Let's initialize these plain data types

**Review by trflynn89 — COMMENTED** [review]
One small thing, otherwise lgtm

File: Userland/Libraries/LibGfx/ImageFormats/WebPWriter.cpp (hunk 1 above)
Comment (nico) [reply Userland/Libraries/LibGfx/ImageFormats/WebPWriter.cpp]: 😬

**Pull request merged**